### PR TITLE
accept and store json endpoint cert metadata on deploy

### PR DESCRIPTION
**Pull request by andreer**
also refactor from tlsSecretKeys -> several "endpoint certificate" classes

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateRetriever.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    public EndpointCertificateRetriever(SecretStore secretStore) {
+        this.secretStore = secretStore;
+    }
+
+    public Optional<EndpointCertificateSecrets> readEndpointCertificateSecrets(EndpointCertificateMetadata metadata) {
+        return Optional.of(readFromSecretStore(metadata));
+    }
+
+    private EndpointCertificateSecrets readFromSecretStore(EndpointCertificateMetadata endpointCertificateMetadata) {
```
Comment (tokle) [inline configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateRetriever.java]: We've had cases where the version of private key and cert were incompatible. Would it make sense to validate that the cert is indeed issued using the private key?

File: configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateMetadataSerializer.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    //          such that what is serialized on version N+1 can be read by version N:
+    //          - ADDING FIELDS: Always ok
+    //          - REMOVING FIELDS: Stop reading the field first. Stop writing it on a later version.
+    //          - CHANGING THE FORMAT OF A FIELD: Don't do it bro.
+
+    private final static String keyNameField = "keyName";
+    private final static String certNameField = "certName";
+    private final static String versionField = "version";
+
+    public static void toSlime(EndpointCertificateMetadata metadata, Cursor object) {
```
Comment (tokle) [inline configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateMetadataSerializer.java]: Won't this violate the principle of "serialized on version N+1 can be read by version N"?

**Review by tokle — CHANGES_REQUESTED** [review]

File: configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateRetriever.java (hunk 1 above)
Comment (andreer) [reply configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateRetriever.java]: Yes, I think I have a ticket on this too. Might as well add it now.

File: configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateMetadataSerializer.java (hunk 2 above)
Comment (andreer) [reply configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateMetadataSerializer.java]: The configserver can already read this format: https://github.com/vespa-engine/vespa/blob/master/configserver/src/test/java/com/yahoo/vespa/config/server/tenant/TlsSecretsKeysTest.java#L49-L58

We only start writing it now.

**Review by tokle — APPROVED** [review]

**Review by bjorncs — APPROVED** [review]
LGTM changes to security-utils

**Pull request merged**